### DISCUSSION TITLE
Map layout

### DISCUSSION THREAD

**Discussion by 3quarterscolon** [TheMowgliMan/Country-Clash#3] [New Features]
We should probably figure out what style we’re going for with the world map. Are we going to include geographic data as well as the borders, or just display a political map? What sort of theme are we going for?

**Comment by TheMowgliMan** [comment]
We're using a tilemap-so I (or somebody else) draw a set of tiles, and build a map with that. That's how OpenTTD gets that tiled look.

We're going to have a geographic and political maps. Resources don't have to be marked as we can just assign a tile a set of resources to be gotten from it, and the player can see what resources are there by somehow opening a menu describing that tile, probably by clicking on it. We don't, however, have to draw another set of tiles for political maps, as we can use shaders (For what we're doing, basically customizing the color of the tile) to do that.

**Reply by 3quarterscolon** [reply]
By “tile”, do you mean square? because borders aren’t going to be straight lines, so you’d need a separate province map anyway.

**Reply by TheMowgliMan** [reply]
Yes. Theoretically, we could use really tiny tile (>4 px on edge) to give the appearance of smoothed map. Hmmm.... I need to look into this a little more. Maybe we could use a giant texture to draw on the map? And we update the texture as needed? That seems like it's going to be really slow. Hmm...

**Reply by 3quarterscolon** [reply]
My idea was to subdivide the map into regions, but the problem with that is that it requires more effort to make.

**Reply by TheMowgliMan** [reply]
Yeah. Then again, the map is just a giant list of numbers, and we could add regions as needed from the generated list.

**Reply by 3quarterscolon** [reply]
Well, if you have arbitrary region shapes, you have to make sure they line up properly. It probably isn’t too hard, but having never used Godot before, I can’t say that for certain.

**Comment by TheMowgliMan** [comment]
Regions are just squares. We draw the region on that square.

**Reply by 3quarterscolon** [reply]
What do you mean?

**Comment by TheMowgliMan** [comment]
Computer hardware loves squares. A region can be an area that is akin to a
square piece of paper. Stich multiple regions together and you get a
continuous scrollable map.

**Reply by 3quarterscolon** [reply]
So how does that fix the problem of square borders?

**Reply by TheMowgliMan** [reply]
You draw a piece of map onto each square. We can also just use a giant 16k (16384*16384) texture to hold the entire map. I don't think we'ed ever need a bigger map then that anyway.

**Reply by TheMowgliMan** [reply]
Like piecing together pieces of a map in real life to get one big map.

**Reply by 3quarterscolon** [reply]
I know that that works with a topographical or satellite-esque map, but how will regions work with that?

**Reply by 3quarterscolon** [reply]
Like, how do you display the area that a country owns?

**Reply by TheMowgliMan** [reply]
By tinting that part of the map with a color that represents the country.

**Reply by 3quarterscolon** [reply]
but if the parts are divided into squares, wouldn’t that result in square borders again?

**Reply by 3quarterscolon** [reply]
I’m sorry if I’m being rude. I’m sure you know what you’re doing, but I just want to know how this is going to be implemented.

**Reply by TheMowgliMan** [reply]
I'll show you a general idea with screenshots soon.  We could also not divide the map at all, it seems to perform fine.

**Reply by 3quarterscolon** [reply]
I’m not talking about the actual color map, I’m asking how country borders are going to be drawn. Because they have to update at some point, and I don’t know how.

**Reply by TheMowgliMan** [reply]
They're going to be redrawn when the rest of the map redraws- every 60 frames (Faster if I can manage it).

**Reply by 3quarterscolon** [reply]
Are they going to be made of tiny squares?

**Reply by TheMowgliMan** [reply]
As all textures are, yes. However, as https://github.com/TheMowgliMan/Country-Clash/discussions/5 demonstrates, they are typically not visible.

**Reply by 3quarterscolon** [reply]
Okay, thanks. That’s all I wanted to know. (Now that I think about it, I have one more question. How are you going to be able to inspect the squares if they’re that small?)

**Reply by TheMowgliMan** [reply]
Your welcome.

**Reply by TheMowgliMan** [reply]
How's work coming on the main menu?

**Reply by 3quarterscolon** [reply]
I’m still working on homework :( sorry

**Reply by 3quarterscolon** [reply]
The internet broke for a while this week so I’m way behind.

**Reply by TheMowgliMan** [reply]
Eh, it's fine. I still have plenty to do on the back end anyway.
